### ISSUE TITLE
Dead quantile code in SmartNoise-SQL is vulnerable to floating-point attacks

### ISSUE THREAD

**Issue by TedTed** [opendp/smartnoise-sdk#572]
Hi folks,

[This code](https://github.com/opendp/smartnoise-sdk/blob/main/sql/snsql/sql/_mechanisms/approx_bounds.py) contains a `quantile` function which is implemented naively. It's vulnerable to fairly trivial floating-point vulnerabilities. For example, the following code:

```py
import snsql
from snsql.sql._mechanisms.approx_bounds import quantile
print(quantile([0]*1491+[1], 0.5, 1, 0, 1))
```

will always return a value, but the following code:

```py
import snsql
from snsql.sql._mechanisms.approx_bounds import quantile
print(quantile([0]*1492+[1], 0.5, 1, 0, 1))
```

will always crash with "ValueError: probabilities contain NaN".

I haven't tried it, but I'm also fairly certain that it's vulnerable to [precision-based attacks](https://arxiv.org/pdf/2207.13793.pdf), for the same reason as diffprivlib and SmartNoise Core: the use of `np.random.uniform` is very dangerous in that context.

Thankfully, this code doesn't seem to be actually used anywhere (that I can see). Nonetheless, it probably seems worth removing it from the repository.

**Comment by joshua-oss** [comment]
Thanks for reporting. This is, indeed, dead and buggy code, and we will remove it. The intent was to use this to compute quantiles when we have a reliable cross-engine way to sample row values (rather than just return aggregates). When we reach that point, we'll use an algorithm that is safe, and ensure that our implementation doesn't leak values.

**Comment by joshua-oss** [comment]
Fixed in 1.0.1
